### PR TITLE
IA-2343 use blessed image

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-2343

1. use blessed image
2. update script to support bumping major version

### CODE REVIEW

File: terra-jupyter-base/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # adapted from https://hub.docker.com/r/jupyter/base-notebook/ AKA https://github.com/jupyter/docker-stacks/tree/master/base-notebook
 
-FROM marketplace.gcr.io/google/ubuntu1804:latest
+FROM us.gcr.io/broad-dsp-gcr-public/base/python:debian
```
Comment (rtitle) [inline terra-jupyter-base/Dockerfile:3]: Does this change from Ubuntu to Debian?

Comment (Qi77Qi) [reply terra-jupyter-base/Dockerfile:3]: yea...debian is the closest distro among [blessed images](https://github.com/broadinstitute/dsp-appsec-blessed-images#dsp-appsec-blessed-base-images)

**Comment by rtitle** [comment]
I'm a little nervous to bump the base image for _all_ Jupyter images TBH -- before merging we should build all images and do a lot of testing.

Is there a strong reason we don't want to use `marketplace.gcr.io/google/ubuntu*` images @dinvlad? I think we chose a Google-maintained base image based on a recommendation.. can't remember from who exactly, could try to dig it up.

Also is there a "blessed" Ubuntu image we could use?

**Comment by rtitle** [comment]
Also: does `terra-docker` fall under the AppSec requirement of "blessed" images? These images don't run in Terra infrastructure per se, but are purely launched/managed/access controlled by Leo in users' Terra billing projects. Just wondering what the actual requirement is.

**Comment by dinvlad** [comment]
@rtitle good point here! we don't have a _strict_ requirement for scanning end-user images; however, since those images are used universally by users throughout Terra, _and_ they do run user-facing _services_, they seemed like a good candidate to scan. So if we scan them, this will raise fewer questions from the auditors later on 😉

**Comment by dinvlad** [comment]
Also re `marketplace.gcr.io/google/ubuntu*` or any other base images, you're free to use those! The only issue with "Marketplace" GCR images is that access to them is granted to `allAuthenticatedUsers`, instead of `allUsers`. This means that one has to have a Google identity to authenticate with GCR first, in order to pull them. It's not possible to pull them anonymously, like the regular `Dockerfile` inside our GitHub workflow does. So to access those, we'd need to supply a Service Account key to the workflow. If that would work for you, we can do that.

**Comment by rtitle** [comment]
Makes sense re scanning, thanks.

I'm not necessarily married to the GCP Marketplace images, but since these are user-facing images I would just like to avoid a lot of headache involved in swapping out the OS/base image. E.g. who knows if this would break some user's code by changing some python/system dependency. I think another reason we used `marketplace.gcr.io/google/ubuntu1804:latest` is for GCP to manage security patches/etc for us. Not sure what trivy thinks though. :)

**Comment by dinvlad** [comment]
Yep totally understandable! So let's sync offline how you could set up a dedicated SA key for this, and I'll help with the setup in the workflow (already tested the Yaml config for this, to be added)

**Comment by dinvlad** [comment]
After thinking a bit more about this use case, you're absolutely right @rtitle that it's not worth the additional effort to set up scanning of these images for user-facing containers, as the impact would be low in our threat model anyways (unlike for regular services).

Please feel free to reverse the PR to remove Trivy from this repo!

**Comment by rtitle** [comment]
Ok great, thanks @dinvlad !

**Comment by Qi77Qi** [comment]
hmm..I was going to update this branch

**Pull request closed without merging**